### ISSUE TITLE
error on parameter decorator in overload signature 

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1125]
```typescript
function q(...args) {}
class C {
  // @q is invalid. This is an overload signature. quick-lint-js should diagnose this.
  f(a, @q b)
  // @q is okay. This is the main method.
  f(param1, @q param2) {}
}
```
